datafusion/core/tests/parquet/arrow_statistics.rs

```diff
@@ -0,0 +1,652 @@
+            1577841061000000000,
+            1578704521000000000,
+        ])),
+        // maxes are [1577926861000000000, 1578704461000000000, 157879086100000000]
+        expected_max: Arc::new(Int64Array::from(vec![
+            1577926861000000000,
+            1578704461000000000,
+            1578790861000000000,
+        ])),
+        // nulls are [1, 2, 1]
+        expected_null_counts: UInt64Array::from(vec![1, 2, 1]),
+        // row counts are [8, 8, 4]
+        expected_row_counts: UInt64Array::from(vec![8, 8, 4]),
+    }
+    .run("nanos");
+
+    // micros
+    let reader = parquet_file_many_columns(Scenario::Timestamps, row_per_group).await;
+    Test {
+        reader,
+        expected_min: Arc::new(Int64Array::from(vec![
+            1577840461000000,
+            1577841061000000,
+            1578704521000000,
+        ])),
+        expected_max: Arc::new(Int64Array::from(vec![
+            1577926861000000,
+            1578704461000000,
+            1578790861000000,
+        ])),
+        expected_null_counts: UInt64Array::from(vec![1, 2, 1]),
+        expected_row_counts: UInt64Array::from(vec![8, 8, 4]),
+    }
+    .run("micros");
+
+    // millis
+    let reader = parquet_file_many_columns(Scenario::Timestamps, row_per_group).await;
+    Test {
+        reader,
+        expected_min: Arc::new(Int64Array::from(vec![
+            1577840461000,
+            1577841061000,
+            1578704521000,
+        ])),
+        expected_max: Arc::new(Int64Array::from(vec![
+            1577926861000,
+            1578704461000,
+            1578790861000,
+        ])),
+        expected_null_counts: UInt64Array::from(vec![1, 2, 1]),
+        expected_row_counts: UInt64Array::from(vec![8, 8, 4]),
+    }
+    .run("millis");
+
+    // seconds
+    let reader = parquet_file_many_columns(Scenario::Timestamps, row_per_group).await;
+    Test {
+        reader,
+        expected_min: Arc::new(Int64Array::from(vec![
+            1577840461, 1577841061, 1578704521,
+        ])),
+        expected_max: Arc::new(Int64Array::from(vec![
+            1577926861, 1578704461, 1578790861,
+        ])),
+        expected_null_counts: UInt64Array::from(vec![1, 2, 1]),
+        expected_row_counts: UInt64Array::from(vec![8, 8, 4]),
+    }
+    .run("seconds");
+}
+
+// date with different row group sizes
+#[tokio::test]
+async fn test_dates_32_diff_rg_sizes() {
+    let row_per_group = 13;
+
+    // This creates a parquet files of 3 columns named "date32", "date64", "names"
+    // "date32" --> Date32Array
+    // "date64" --> Date64Array
+    // "names" --> StringArray
+    //
+    // The file is created by 4 record batches (each has a null row), each has 5 rows but then will be split into 2 row groups with size 13, 7
+    let reader = parquet_file_many_columns(Scenario::Dates, row_per_group).await;
+
+    Test {
+        reader,
+        // mins are [18262, 18565,]
+        expected_min: Arc::new(Int32Array::from(vec![18262, 18565])),
```

Comment:
I would actually expect the returned type to be `Date32Array` as the underlying arrow type is Date32 -- I don't think we need to make this change as part of this PR

Reply:
https://github.com/apache/datafusion/issues/10587